python/ray/tests/test_basic.py

```diff
@@ -2536,26 +2534,53 @@ def __exit__(self, exc_type, exc_value, traceback):
 def test_logging_to_driver(shutdown_only):
     ray.init(num_cpus=1, log_to_driver=True)
 
+    bound_l = 1
+    bound_h = 9
+
+    # Randomize output
+    j = random.randint(bound_l, bound_h)
+
     @ray.remote
-    def f():
+    def f(j):
         # It's important to make sure that these print statements occur even
         # without calling sys.stdout.flush() and sys.stderr.flush().
-        for i in range(100):
-            print(i)
-            print(100 + i, file=sys.stderr)
+        for i in range(j * 100, (j + 1) * 100 - 50):
+            print(-i)
+            print(-(50 + i), file=sys.stderr)
 
     captured = {}
     with CaptureOutputAndError(captured):
-        ray.get(f.remote())
+        ray.get(f.remote(j))
         time.sleep(1)
 
     output_lines = captured["out"]
-    for i in range(200):
-        assert str(i) in output_lines
+    for i in range(bound_l * 100, (bound_h + 1) * 100):
+        if i >= j * 100 and i < (j + 1) * 100:
+            assert str(-i) in output_lines
+        else:
+            assert str(-i) not in output_lines
     error_lines = captured["err"]
     assert len(error_lines) == 0
 
 
+def test_logging_to_multiple_drivers(shutdown_only):
+    file_name = os.path.abspath(__file__) + "::test_logging_to_driver"
```

Comment:
Interesting approach. I have some suggestions for making it more consistent with the other tests.

One minor issue is that `python` may not actually find the right `python` command, e.g., if the user started the test using `python3` or something like that. To handle this kind of thing in other places in the codebase, we use `sys.executable`.

Take a look at how the tests are written in e.g., https://github.com/ray-project/ray/blob/master/python/ray/tests/test_multi_node.py

- This uses the `call_ray_start` pytest fixture to start a Ray cluster in the background.
- Then we can connect the test as a driver to the cluster with `ray.init(redis_address=...)`
- Then we can start several drivers in the background that do some printing (both to stdout and stderr) using `run_string_as_driver` or `run_string_as_driver_nonblocking` (2 or three should be sufficient).
- We can capture all of the main test scripts stdout/stderr using `with CaptureOutputAndError` like in `test_logging_to_driver`.
- We can run some tasks from the main script that do some logging.
- We can make sure that we do see the output from the tasks that the main script submitted. And make sure that we don't see the output from the tasks that the background drivers submitted.